lib/pact/command/cont_command.ex

```diff
@@ -0,0 +1,333 @@
+          signers: signers()
+        }
+
+  defstruct [
+    :network_id,
+    :data,
+    :nonce,
+    :meta_data,
+    :pact_tx_hash,
+    :step,
+    :proof,
+    :rollback,
+    :signers,
+    keypairs: []
+  ]
+
+  @impl true
+  def new(opts \\ nil)
+
+  def new(opts) when is_list(opts) do
+    network_id = Keyword.get(opts, :network_id)
+    data = Keyword.get(opts, :data)
+    nonce = Keyword.get(opts, :nonce, "")
+    meta_data = Keyword.get(opts, :meta_data, %MetaData{})
+    pact_tx_hash = Keyword.get(opts, :pact_tx_hash, "")
+    step = Keyword.get(opts, :step, %Step{})
+    proof = Keyword.get(opts, :proof)
+    rollback = Keyword.get(opts, :rollback, %Rollback{})
+    keypairs = Keyword.get(opts, :keypairs, [])
+    signers = Keyword.get(opts, :signers, %SignersList{})
+
+    %__MODULE__{}
+    |> set_network(network_id)
+    |> set_data(data)
+    |> set_nonce(nonce)
+    |> set_metadata(meta_data)
+    |> add_keypairs(keypairs)
+    |> add_signers(signers)
+    |> set_pact_tx_hash(pact_tx_hash)
+    |> set_proof(proof)
+    |> set_step(step)
+    |> set_rollback(rollback)
+  end
+
+  def new(_opts), do: %__MODULE__{}
+
+  @impl true
+  def set_network(%__MODULE__{} = cmd_request, network) do
+    case NetworkID.new(network) do
+      %NetworkID{} = network_id -> %{cmd_request | network_id: network_id}
+      {:error, reason} -> {:error, [network_id: :invalid] ++ reason}
+    end
+  end
+
+  def set_network({:error, reason}, _network), do: {:error, reason}
+
+  @impl true
+  def set_data(%__MODULE__{} = cmd_request, data) do
+    case EnvData.new(data) do
+      %EnvData{} -> %{cmd_request | data: data}
+      error -> error
+    end
+  end
+
+  def set_data({:error, reason}, _data), do: {:error, reason}
+
+  @impl true
+  def set_nonce(%__MODULE__{} = cmd_request, nonce) when is_binary(nonce),
+    do: %{cmd_request | nonce: nonce}
+
+  def set_nonce(%__MODULE__{}, _nonce), do: {:error, [nonce: :not_a_string]}
+  def set_nonce({:error, reason}, _nonce), do: {:error, reason}
+
+  @impl true
+  def set_metadata(%__MODULE__{} = cmd_request, %MetaData{} = meta_data),
+    do: %{cmd_request | meta_data: meta_data}
+
+  def set_metadata(%__MODULE__{} = cmd_request, meta_data) do
+    case MetaData.new(meta_data) do
+      %MetaData{} = meta_data -> %{cmd_request | meta_data: meta_data}
+      {:error, reason} -> {:error, [meta_data: :invalid] ++ reason}
+    end
+  end
+
+  def set_metadata({:error, reason}, _metadata), do: {:error, reason}
+
+  @impl true
+  def add_keypair(%__MODULE__{keypairs: keypairs} = cmd_request, %KeyPair{} = keypair),
+    do: %{cmd_request | keypairs: keypairs ++ [keypair]}
+
+  def add_keypair(%__MODULE__{} = cmd_request, keypair) do
+    case KeyPair.new(keypair) do
+      %KeyPair{} = keypair -> add_keypair(cmd_request, keypair)
+      {:error, reason} -> {:error, [keypair: :invalid] ++ reason}
+    end
+  end
+
+  def add_keypair({:error, reason}, _keypair), do: {:error, reason}
+  @impl true
```

Comment:
I think a line break here should be good to separate the implementation from the previous one.